### PR TITLE
Add CD jobs for `ack-chart`

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
Adds the continuous deployment scripts for the `ack-chart` Helm chart.

The `update-chart.sh` script does the following:
- Checks for differences between the latest tag and the current listed version for each subchart
- Updates the subchart dependencies
  - Adding new dependencies for newly released charts
  - Updating version numbers for upgraded charts
- Updates the version number given the set of upgrades/additions
- Pushes the changes to `main` and tags `main` with the newest Chart version number

The `upload-chart.sh` script does the following:
- Builds and uploads the latest version to the ECR public repository

This PR includes 2 new types of jobs in Prow:
- When pushing any controller with a tag, run the `update-chart.sh` script with the relevant Github token details
- When pushing the `ack-chart` with a tag (as `update-chart.sh` does), run the `upload-chart.sh` script

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jljaco — APPROVED** [review]

File: cd/ack-chart/update-chart.sh [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    echo "$ecr_pw" | helm registry login -u AWS --password-stdin public.ecr.aws
+
+    helm dependency update
+}
+
+_add_chart_values_section() {
+    local __service_name=$1
+
+    SERVICE_NAME=$__service_name yq --inplace '.[env(SERVICE_NAME)+"-chart"] += {
+        "enabled": false
```
Comment (jaypipes) [inline cd/ack-chart/update-chart.sh:223]: any reason not to enable the chart by default?

**Review by jaypipes — APPROVED** [review]
@RedbackThomson really nifty Bash-fu here, thank you!

File: cd/ack-chart/update-chart.sh (hunk 1 above)
Comment (RedbackThomson) [reply cd/ack-chart/update-chart.sh:223]: Every chart needs IRSA and other configurations enabled, anyway. I figure if you're going to the trouble of adding those, you can enable it in the same values set

**Review by A-Hilaly — APPROVED** [review]
Great stuff! Can't wait to see this in action :)
/lgtm

**Pull request merged**